cpp/src/datetime/datetime_ops.cu

```diff
@@ -127,6 +127,23 @@ struct extract_day_num_of_year {
   }
 };
 
+// Extract the the quarter to which the timestamp belongs to
+struct extract_quarter_op {
+  template <typename Timestamp>
+  CUDA_DEVICE_CALLABLE int16_t operator()(Timestamp const ts) const
```

Comment:
`uint8_t` should be sufficient?

Reply:
I have no idea why int16 is used here? Why not just int32 for compatability? As mentioned earlier, using int16 seems to be suboptimal.

Reply:
Could you point me to your mentioning that int16 is suboptimal? AFAICT, most other ops in `datetime` uses int16, but for no obvious reasons. 

My point here is that the return type should be in consistent of actual data it's housing. If there are only 4 possible return values, then smallest number of bytes to store it is 1. This is analogous to picking the right underlying type for `enum class`.

Reply:
Here: https://github.com/rapidsai/cudf/pull/8779#discussion_r673615960
The results will be used for some purposes (like compare/add/mult etc... with something else, which typically of int32 type). If you have different data types, you may have trouble due to implicit conversion.

Reply:
In that case, should I be using int32 or int8?

Reply:
Typically, you should better use int32 for safety reasons unless using int8 offers more benefits. That also depends on situations. What is your use case?

Reply:
I think you are right in that the result of this function will be used for compare/add/mult. In that case, I will change this to int32 then. cc @isVoid

Reply:
It seems to be a tricky question whether we should use int32 (for compatibility) or int8 (for performance). @jrhemstad Do you have any idea for this matter?

Reply:
Also, if we do decide to switch to int32 or int8. A lot of the previous datatime ops all used int16 type, in that case should those also be switched to int32 or int8 (assuming it can be switched)?

Reply:
If other APIs already used int16, then just stick with int16 for now, to avoid messing up the existing code and enforce consistency.